mobile_verifier/src/boosting_oracles/mod.rs

```diff
@@ -1,10 +1,305 @@
+    boosting_oracles::assignment::urbanization_multiplier,
+    coverage::SignalLevel,
+    geofence::{Geofence, GeofenceValidator},
+};
 pub use assignment::Assignment;
+use chrono::{DateTime, Duration, Utc};
+use file_store::{file_sink::FileSinkClient, traits::TimestampEncode, FileStore};
+use futures_util::{Stream, StreamExt, TryFutureExt, TryStreamExt};
+use helium_proto::services::poc_mobile as proto;
 use hextree::disktree::DiskTreeMap;
+use rust_decimal::prelude::ToPrimitive;
+use rust_decimal_macros::dec;
+use sqlx::{FromRow, PgPool, QueryBuilder};
+use task_manager::ManagedTask;
+use tokio::{fs::File, io::AsyncWriteExt, sync::Mutex};
+pub use urbanization::Urbanization;
+use uuid::Uuid;
+
+pub trait DataSet {
+    const TYPE: DataSetType;
+
+    fn timestamp(&self) -> Option<DateTime<Utc>>;
+
+    fn update(&mut self, path: &Path, time_to_use: DateTime<Utc>) -> anyhow::Result<()>;
+
+    fn assign(&self, hex: u64) -> anyhow::Result<Assignment>;
+}
+
+pub struct DataSetDownloaderDaemon {
+    pool: PgPool,
+    // Later: footfall and landtype
+    urbanization: Arc<Mutex<Urbanization<DiskTreeMap, Geofence>>>,
+    store: FileStore,
+    oracle_boosting_sink: FileSinkClient,
+}
+
+#[derive(FromRow)]
+pub struct NewDataSet {
+    file_name: String,
+    time_to_use: DateTime<Utc>,
+    status: DataSetStatus,
+}
+
+#[derive(Copy, Clone, sqlx::Type)]
+#[sqlx(type_name = "data_set_status")]
+#[sqlx(rename_all = "lowercase")]
+pub enum DataSetStatus {
+    Pending,
+    Downloaded,
+    Processed,
+}
+
+impl DataSetStatus {
+    pub fn is_downloaded(&self) -> bool {
+        matches!(self, Self::Downloaded)
+    }
+}
+
+impl ManagedTask for DataSetDownloaderDaemon {
+    fn start_task(
+        self: Box<Self>,
+        shutdown: triggered::Listener,
+    ) -> futures::prelude::future::LocalBoxFuture<'static, anyhow::Result<()>> {
+        let handle = tokio::spawn(self.run(shutdown));
+        Box::pin(
+            handle
+                .map_err(anyhow::Error::from)
+                .and_then(|result| async move { result.map_err(anyhow::Error::from) }),
+        )
+    }
+}
+
+impl DataSetDownloaderDaemon {
+    pub fn new(
+        pool: PgPool,
+        urbanization: Arc<Mutex<Urbanization<DiskTreeMap, Geofence>>>,
+        store: FileStore,
+        oracle_boosting_sink: FileSinkClient,
+    ) -> Self {
+        Self {
+            pool,
+            urbanization,
+            store,
+            oracle_boosting_sink,
+        }
+    }
+
+    pub async fn run(self, shutdown: triggered::Listener) -> anyhow::Result<()> {
+        // Another option I considered instead of polling was to use ENOTIFY, but that seemed
+        // not as good, as it would hog a pool connection.
+        //
+        // We set the poll duration to 30 minutes since that seems fine.
+        let poll_duration = Duration::minutes(30);
+
+        loop {
+            // Find the latest urbanization file
+            tracing::info!("Checking for new data sets");
```

Comment:
Be good to push this chunk of code  to determine and download the file  out to its own function